contracts/utils/TimestampedHashRegistry.sol

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/cryptography/EIP712.sol";
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import "./interfaces/ITimestampedHashRegistry.sol";
+
+// TODO: would SelfMulticall be useful?
+contract TimestampedHashRegistry is Ownable, EIP712, ITimestampedHashRegistry {
+    using ECDSA for bytes32;
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    mapping(bytes32 => EnumerableSet.AddressSet) private _hashTypeToSigners;
+
+    mapping(bytes32 => SignedHash) public hashTypeToSignedHash;
+
+    bytes32 private constant _SIGNED_HASH_TYPE_HASH =
+        keccak256(
+            "SignedHash(bytes32 hashType,bytes32 hash,uint256 timestamp)"
+        );
+
+    constructor() EIP712("TimestampedHashRegistry", "1.0.0") {}
+
+    function setHashTypeSigners(
```

Comment:
The only point to use `EnumerableSet.AddressSet` for signers would be to have `addSigner()`/`removeSigner()` functions. If we're going to have to set/reset all signers, keeping them in `address[]` as in DapiFallbackV1 is much cheaper.

Reply:
That being said, `addSigner()`/`removeSigner()` is neat (but would require SelfMulticall to be inherited)

Reply:
Right, I thought about it but then I didn't want to deviate too much from DapiFallbackV1 implemenation. I'll add these functions instead 👍🏻

Reply:
I might be also useful to have a `resetSigners()` function to clear all signers for a hash type name 🤔

Reply:
> How about adding a signers array to the constructor arguments and do a initial set while deploying like in [DapiFallbackV1](https://github.com/api3dao/manager-multisig/blob/main/chain/contracts/DapiFallbackV1.sol#L93)? Then one can use add/remove functions later on to modify the set.

I normally avoid that for the deterministic deployments to have the same address. It's better to have an [initalization function](https://github.com/safe-global/safe-contracts/blob/main/contracts/base/OwnerManager.sol#L28) behind `onlyOwner`.

> I might be also useful to have a resetSigners() function to clear all signers for a hash type name 🤔 (or a setSigners(address[] signers) that first deletes all signers and then sets based on the signers param)

If [GnosisSafe](https://github.com/safe-global/safe-contracts/blob/main/contracts/base/OwnerManager.sol) doesn't need set/reset functions, neither do we imo.